### PR TITLE
:book: adding investigation for Edge Multicluster

### LINKED ISSUE #2392: feature: KCP.io Investigation Document for Edge Multicluster
### Feature Description

We would like to represent the Edge MC work/sub-project in KCP.io as a KCP sub-project.

### Proposed Solution

adding an Edge MC investigation MD

### Alternative Solutions

_No response_

### Want to contribute?

- [X] I would like to work on this issue.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by clubanderson**
## Summary
add investigation document for Edge Multicluster

## Related issue(s)

Fixes #2392

### CODE REVIEW

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+Kubernetes offers a state-based management platform to declaratively developing and operating workloads. Kube allows a developer to
+ create workloads that are node-agnostic and can be constrained to operate in places where certain conditions are met. Operators enjoy
+  gaurantee that workloads are rescheduled to work wherever constraints allow them to run. This approach has satisfied the large
+   majority of use cases in the cloud native community.
+
+There is a growing community of cloud native developers and operators that work for organizations that require more fine-grain control
+ over the distribution of workload due to the nature of their business. Telecommunications, industrial manufacturers, and automotive
+  companies would like to rely on Kubernetes state-based management but they need more flexibility to accomodate for conditions like
+   large-scale deployments, disconnected and autonomous operation, and constrained cluster size. These requirements represent the Edge
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:17]: large-scale deployments is not really a "condition" , it's more of a requirement that impact the underlying solution/platform

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+weight: 1
+description: >
+  Investigation of edge multi-cluster
+---
+
+Kubernetes offers a state-based management platform to declaratively developing and operating workloads. Kube allows a developer to
+ create workloads that are node-agnostic and can be constrained to operate in places where certain conditions are met. Operators enjoy
+  gaurantee that workloads are rescheduled to work wherever constraints allow them to run. This approach has satisfied the large
+   majority of use cases in the cloud native community.
+
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:13]: The relations between the above k8s paragraph and the rest of the doc is not clear to me. I think that in k8s we mainly talk about placing pods on nodes, which also includes constrains, rescheduling, etc..  
In KCP we are talking bout placing workloads on clusters which is different level (e.g., in KCP we don't want to expose Node capabilities or per Pod scheduling)

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+There is a growing community of cloud native developers and operators that work for organizations that require more fine-grain control
+ over the distribution of workload due to the nature of their business. Telecommunications, industrial manufacturers, and automotive
+  companies would like to rely on Kubernetes state-based management but they need more flexibility to accomodate for conditions like
+   large-scale deployments, disconnected and autonomous operation, and constrained cluster size. These requirements represent the Edge 
+   Multi-cluster use case.
+
+A key area of investigation for `kcp` is exploring large-scale deployment of workloads that can operate under constraints and
+ conditions unlike those accomodated by upstream Kubernetes. Ideally, we want declaritively defined workloads to operate across a
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:21]: Why do we compare to upstream k8s, this is a proposal for KCP, so maybe "unlike those accommodated by existing KCP TMC" ?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## Goal: Placement strategies
+
+There are many different ways to characterize deployments of workloads in the Edge use case. Support for blue/green, canary, carbon
+ intensity sensitive, and threat avoidance are some of the deployment strategies to consider when casting out 100's or 1000's of
+  workloads to millions of destinations. Having the ability to assign workloads using a declaritively defined placement strategy to
+   edge cluster locations, even if they are offline at times, is a key requirement of the edge multi-cluster use case.
+
+## Overlap of roles, responsibilities, integrations, and security
+
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:34]: Can we have an example for something that is not covered by the existing KCP roles?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+   community contribution powered by integration and proper security features. All of these considerations must work across an evolving
+    hierarchy that combines small and large control plane API services.
+
+## Hierarchy and Scale are core considerations
+
+KCP has introduced the concept of transparent multi-cluster which can be used to expand the number of entry points for workloads and
+ operations by reducing the size of the control plane to a single binary. Synchronization between KCP virtual workspaces and physical
+  clusters has made it possible to deploy workloads centrally and then fan out to any defined subset of clusters. We would like to
+   leverage multiple levels of hierarchy (cascading multiple instances of KCP and Kubernetes cluster) to arrive at a distributed system
+    capable of reaching any targeted edge cluster with/without edge devices attached. Building out such a hierarchy will require a
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:47]: Not sure the scale should be linked to the hierarchical KCP architecture. Scale need to be explored because the large scale nature (millions of targets) of the use case. Actually hierarchical architecture may be a possible solution for some of the scaling issues (including the DB if we can avoid centralized place for data)

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+     scalable database to hold the objects referenced by each of their associated API servers. The default database for KCP and
+      Kubernetes is etcd. We will investigate alternatives to etcd distribution and replacement databases to increase scalability with
+       sacrificing performance and resilience without negatively impacting cost.
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:50]: I think scale is a general issue, maybe put the etcd as one example of the general area.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 2 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:13]: the link is in the first line of the next paragraph - there are more organizations looking for more detailed control over where a workload runs then what is currently available in off-the-shelf Kubernetes.  To me, using KCP and developing KCP-Edge helps us run workloads at multiple locations (large-scale), in a way that is disconnect-tolerant, and perhaps even without intervention (autonomy).  Today, we can't do much of this because the underlying Kube abstractions don't allow for this.  Therefore we need new parameters, abstractions, and controllers to act in a way that compliments edge use-cases (scale, disconnect-tolerant, autonomous).

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 1 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:17]: I can change the term "conditions" to "requirements" if you like.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 3 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:21]: I added "or KCP's Transparent MultiCluster (TMC)" to the statement

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 4 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:34]: can you give me an example of another case other than the smartphone analogy?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 2 above)
Comment (ezrasilvera) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:13]: I think I didn't clarify my point very well.. Let me try again. In general this is a proposal for exploration under KCP so the preface should explain how the proposed use case is related to KCP and then what are the specific areas under KCP that need to be explored to address special requirements of the Edge use case. 
In addition, I think the k8s paragraph includes statements that are not fully clear or accurate (k8s is aware of nodes, it's doing pods based scheduling and not "workload" into "places", as well as  the constraint are pod to Nodes and not complicated workloads). 
Maybe we can put instead a paragraph what people are trying to do with KCP and why, and then show why the Edge use case may be a good fit for this architecture + what are the gaps.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 4 above)
Comment (ezrasilvera) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:34]: I didn't think abut it so I don't have one (which is my here :-)  ). I thought the claim is that the `roles, responsibilities, integrations, and security`  of KCP doesn't address the Edge use case and need to be explored. If this is the claim I think you need to show an example of such a gap

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 2 above)
Comment (MikeSpreitzer) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:13]: @clubanderson : while "require more fine-grain control over the distribution of workload" is not an inaccurate statement about edge, I think it omits a critical point.  This involves expanding beyond a single cluster, and explicitly caring about clusters.  This makes a clear distinction from the TMC work, see how its Investigation page stipulates that clusters are relatively unimportant.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
+---
+title: "Edge multi-cluster"
+linkTitle: "Edge multi-cluster"
+weight: 1
+description: >
+  Investigation of edge multi-cluster
+---
+
+Kubernetes offers a state-based management platform to declaratively developing and operating workloads. Kube allows a developer to
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:9]: s/operating workloads/operating containerized workloads/

**Comment by MikeSpreitzer** [comment]
I find much of the structure of the existing draft confusing.  There is an introductory section, followed by four 2nd-level headings of our own composition, followed by three 2nd-level headings copied from the TMC Investigation page ("Areas of Investigation", "Use Cases", and "Progress").  The relation of the last three 2nd-level headings to each other and the rest of the document is clear enough; it is the earlier material that I have trouble with.  Among the first four 2nd-level headings, two are marked as "Goal" and two are not.  The introductory section should, as mentioned elsewhere in this PR, contrast Edge MC with Transparent MC.  Yet the contrast discussed so far for the introductory section covers much less than the nine points listed in https://docs.google.com/document/d/1vwOJ3264wm17Y8fCZY9bgSjgrkO9md4qofOjUefE1N0 .  And the first four 2nd-level headings continue to draw contrasts with TMC --- although that might not be their main point.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+KCP has introduced the concept of transparent multi-cluster which can be used to expand the number of entry points for workloads and
+ operations by reducing the size of the control plane to a single binary. Synchronization between KCP virtual workspaces and physical
+  clusters has made it possible to deploy workloads centrally and then fan out to any defined subset of clusters. We would like to
+   leverage multiple levels of hierarchy (cascading multiple instances of KCP and Kubernetes cluster) to arrive at a distributed system
+    capable of reaching any targeted edge cluster with/without edge devices attached. Building out such a hierarchy will require a
+     scalable database to hold the objects referenced by each of their associated API servers. The default database for KCP and
+      Kubernetes is etcd. We will investigate alternatives to etcd distribution and replacement databases to increase scalability with
+       sacrificing performance and resilience without negatively impacting cost.
+
+## Goal: Modularity is key
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:52]: The heading of this section is motherhood and apple pie, the first sentence is a vague salute, and the remainder makes a different point.

I do not think we need to advocate modularity as a vague generality.  I think we *should* make it clear somewhere --- probably at the top of the document --- that we are talking about a peer of TMC, those being two layers that rest on the other two lower layers of kcp.

Regarding flexibility of hierarchy: I think we should clearly say that the customer brings a hierarchy that is connected with their business, we do not get to dictate the hierarchy.  Of course we can impose constraints based on what we consider feasible, but these should be no more than really necessary.

Regarding flexibility in deployment and integration: perhaps you are saying that the customer brings their own inventory management solution.  If so, I think that is a better way to say it.  Perhaps you are trying to get at another distinction from TMC, which I still have not found a clear formulation for, but it is about the fact that a customer doing edge computing needs a solution not only for containerized workload but also lower layers of the stack, down to stuff described with words like "infrastructure".

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:34]: I am not sure "Overlap" is the right focus here.  In many ways, the point is about _divisions_ rather than _overlap_.  I usually say this with words like "a more complex arrangement of roles and responsibilities".  I am not sure "integrations" and "security" belong in the same breath.

TMC does not have something like the arrangement of roles and responsibilities that Andy described for smartphones.  Other examples I use:
- personal computer issued to corporate employee
- car with software managed by owner, manufacturer, possibly dealer, possibly other service provider
- a manufacturing operation with many plants with local authority as well as a corporate engineering team and a web of suppliers

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+Today's smartphone is not only the property of the user who purchased it. The reality is that application developers (app store),
+ corporate entities (vpn profiles), and the smartphone manufacturer have roles and responsibilities on these devices. Edge cluster
+  locations are no different in this respect. A scalable Edge multi-cluster solution can thrive and advance if there is support for
+   community contribution powered by integration and proper security features. All of these considerations must work across an evolving
+    hierarchy that combines small and large control plane API services.
+
+## Hierarchy and Scale are core considerations
+
+KCP has introduced the concept of transparent multi-cluster which can be used to expand the number of entry points for workloads and
+ operations by reducing the size of the control plane to a single binary. Synchronization between KCP virtual workspaces and physical
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:44]: "virtual workspace" is a technical term in kcp that I think is _not_ what you mean here.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+  clusters has made it possible to deploy workloads centrally and then fan out to any defined subset of clusters. We would like to
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:45]: s/deploy/define/

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+Today's smartphone is not only the property of the user who purchased it. The reality is that application developers (app store),
+ corporate entities (vpn profiles), and the smartphone manufacturer have roles and responsibilities on these devices. Edge cluster
+  locations are no different in this respect. A scalable Edge multi-cluster solution can thrive and advance if there is support for
+   community contribution powered by integration and proper security features. All of these considerations must work across an evolving
+    hierarchy that combines small and large control plane API services.
+
+## Hierarchy and Scale are core considerations
+
+KCP has introduced the concept of transparent multi-cluster which can be used to expand the number of entry points for workloads and
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:43]: The first sentence is unclear to me.  What is an "entry point for workloads and operations"?  Is that "workloads and operations" trying to get at a noun-verb duality, or is a distinction between dev and ops?  What sort of "entry point" is meant?  Is it something like an endpoint for communication, or a way into thinking about a lot of stuff, or ... ?  What has "reducing the size of the control plane to a single binary" got to do with any of that?  Remember also that (a) the kcp binary is prepared to work with an external etcd cluster and this is the only sensible architecture for production use and (b) by the time we get sharding the picture is more complicated.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 10 above)
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:45]: But we argue that "any" is not really true/practical.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 5 above)
Comment (MikeSpreitzer) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:47]: Yes.
Remember also how scale complicates other aspects of the problem, making simple solutions infeasible.

**Comment by sttts** [comment]
@clubanderson am happy to approve, but needs a lgtm by @MikeSpreitzer

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 1 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:17]: changed to 'large number of edge locations'

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 2 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:13]: I made some edits to the first paragraph to more accurately depict how people use Kube and what they expect as outcomes.  I also changed the second paragraph to express there are situations where placement matters and pods can't just be scheduled anywhere to do work.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 4 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:34]: I removed the word 'integrations'

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 5 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:47]: I remove 'hierarchy' and replaced with 'flexible topology'

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 7 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:9]: replaced

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 8 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:52]: removed the modularity goal altogether.  It is inherent of Kube and we are not changing this, so why mention it.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 10 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:45]: updated

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 11 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:43]: I changed this paragraph to mention kcp's minimal api server and kcp's TMC as two separate concepts.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md (hunk 10 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster copy.md:45]: changed to 'specific'

**Comment by clubanderson** [comment]
@MikeSpreitzer @ezrasilvera can you review

**Comment by clubanderson** [comment]
@MikeSpreitzer @ezrasilvera would like to get this posted this week to help us garner more search hits.  Please help us get this content written and approved.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+Kubernetes offers a widely adopted, state-based management platform to declaratively develop and operate containerized workloads. Kubernetes allows a 
+developer to create deployments that are node-agnostic and can be constrained to operate in places where certain conditions are met. Developers and 
+Operators understand that pods will be rescheduled to execute on nodes wherever constraints allow them to run. This approach has satisfied the large
+   majority of use cases in the cloud native community.
+
+There is a growing community of cloud native developers and operators that require more fine-grained control
+ over the distribution of pods due to the nature of their business. Telecommunications, industrial manufacturers, and automotive
+  companies would like to rely on Kubernetes state-based management but they need more flexibility and specificity to accomodate for requirements like
+  large numbers of edge locations, disconnected and autonomous operation, and constrained cluster size. In the edge multi-cluster usecase, there are
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:17]: I'm not sure that the "fine-grained control" and "need more flexibility" is the incentive for Edge-mc. I would suggest to replace the whole paragraph above and  simply go with something like:  "...  would like to rely on Kubernetes state-based management but have unique requirements that are not covered by Kubernets or KCP, like....

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  situations when pods must run on a specific edge location on a specific node. Managing a set of labels, taints, and tolerations to cover all edge 
+  locations is too difficult.
+
+A key area of investigation for `kcp` is exploring large-scale deployment of workloads that can operate under constraints and
+ requirements unlike those accomodated by upstream Kubernetes or KCP's Transparent MultiCluster (TMC). Ideally, we want declaritively defined workloads
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:22]: "unlike those accomodated by upstream Kubernetes"  this is confusing as there are no workspaces in k8s so obviously they can't be accommodated....

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+   edge cluster locations, even if they are offline at times, is a key requirement of the edge multi-cluster use case.
+
+## Overlap of roles, responsibilities, and security
+
+Today's smartphone is not only the property of the user who purchased it. The reality is that application developers (app store),
+ corporate entities (vpn profiles), and the smartphone manufacturer have roles and responsibilities on these devices. Edge cluster
+  locations are no different in this respect. A scalable Edge multi-cluster solution can thrive and advance if there is support for
+   community contribution powered by integration and proper security features. All of these considerations must work across an evolving
+    hierarchy that combines small and large control plane API services.
+
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:41]: The section is not clear. To me the smartphone analogy was confusing, not sure it's adding info. Why not specify the actual roles you have in mind and what are the challenges in the Edge requirements (I don't have a good example for that ..)

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+## Hierarchy and Scale are core considerations
+
+KCP has introduced the concept of a minimal API server which can be used to expand the number of places workloads can be deployed
+  by reducing the size of the control plane to a single binary. Furthermore, KCP's Transparent Multi-cluster (TMC) has made it possible to define
+  and deploy workloads centrally and then fan them out to specific edge locations. We would 
+   like to leverage different topologies (for instance: cascading multiple instances of KCP and Kubernetes cluster) to arrive at a 
+    distributed system capable of reaching any targeted edge cluster with/without edge devices attached. Topology flexibility 
+     will require a scalable database to hold the objects referenced by each of their associated API servers. The default database for
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:49]: not sure it's the "topology flexibility"  - why not just say that due to the very large scale of edge environments that can reach millions of locations, resulting in the need to manage millions of objects by the kcp based control plan.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  and deploy workloads centrally and then fan them out to specific edge locations. We would 
+   like to leverage different topologies (for instance: cascading multiple instances of KCP and Kubernetes cluster) to arrive at a 
+    distributed system capable of reaching any targeted edge cluster with/without edge devices attached. Topology flexibility 
+     will require a scalable database to hold the objects referenced by each of their associated API servers. The default database for 
+      KCP and Kubernetes is etcd. We will investigate alternatives to etcd distribution and replacement databases to increase 
+       scalability without sacrificing performance and resilience and considering factors which could negatively impact cost.
+
+## Areas of investigation
+
+1. Define the high level user Edge multicluster use case
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:55]: Maybe "Define the high level Edge multicluster use cases"  ?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+2. Explore characteristics of most common Kube workload objects that could allow them to be placed in edge cluster locations
+3. Desired placement expression​: Need a way for one center object to express large number of desired copies​
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:57]: Did we define what is a "center" before this place?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+4. Scheduling/syncing interface​: Need something that scales to large number of destinations​
+5. Rollout control​: Client needs programmatic control of rollout, possibly including domain-specific logic​
+6. Customization: Need a way for one pattern in the center to express how to customize for all the desired destinations​
+7. Status from many destinations​: Center clients may need a way to access status from individual edge copies
+8. Status summarization​: Client needs a way to say how statuses from edge copies are processed/reduced along the way from edge to center​.
+9. Seek consensus in user communities on whether the interfaces are practical
+10. Formalize parts of the prototype into project(s) drawing on the elements above if successful!
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:64]: Not clear

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+## Areas of investigation
+
+1. Define the high level user Edge multicluster use case
+2. Explore characteristics of most common Kube workload objects that could allow them to be placed in edge cluster locations
+3. Desired placement expression​: Need a way for one center object to express large number of desired copies​
+4. Scheduling/syncing interface​: Need something that scales to large number of destinations​
+5. Rollout control​: Client needs programmatic control of rollout, possibly including domain-specific logic​
+6. Customization: Need a way for one pattern in the center to express how to customize for all the desired destinations​
+7. Status from many destinations​: Center clients may need a way to access status from individual edge copies
+8. Status summarization​: Client needs a way to say how statuses from edge copies are processed/reduced along the way from edge to center​.
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:62]: This is due to the issue in (7) - maybe combine or link ?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+9. Seek consensus in user communities on whether the interfaces are practical
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:63]: Maybe, "Seek consensus in relevant user communities for the interfaces definitions" ?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+### As a user
+
+1. I can `kubectl apply` a workload that is agnostic to node placement to `kcp` and see the workload assigned to many edge location
+ resources and start running and the status summarized back to me.
+2. I can move an application (defined in 1) between two workload clusters by changing a single high level attribute
+3. As a user when I move an application (as defined in 2) no disruption of internal or external traffic is visible to my consumers
+4. As a user I can debug my application in a familiar manner regardless of cluster
+5. As a user with a stateful application by persistent volumes can move / replicate / be shared across clusters in a manner consistent
+ with my storage type (read-write-one / read-write-many).
+
```
Comment (ezrasilvera) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:79]: Few more:
- As an edge admin I can define a "template" for group of edge locations and be able to apply rule-based customization for specific edge locations.
- As an admin I can watch a summary of a group of edge locations
- As an admin I can request a detailed status  for specific edge location
- As an admin I can define roles for accessing and managing the edge locations and applications on the edge locations

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 16 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:55]: updated as suggested, thank you

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 18 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:64]: I replaced with this "10. Formalize parts of the prototype into a solution that represents the requirements of edge multicloud use cases."

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+## Areas of investigation
+
+1. Define the high level user Edge multicluster use case
+2. Explore characteristics of most common Kube workload objects that could allow them to be placed in edge cluster locations
+3. Desired placement expression​: Need a way for one center object to express large number of desired copies​
+4. Scheduling/syncing interface​: Need something that scales to large number of destinations​
+5. Rollout control​: Client needs programmatic control of rollout, possibly including domain-specific logic​
+6. Customization: Need a way for one pattern in the center to express how to customize for all the desired destinations​
+7. Status from many destinations​: Center clients may need a way to access status from individual edge copies
```
[the remark this answers is not in the document]
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:61]: replaced with "7. Status from many destinations​: Center clients may need the ability to get and store status of vast number of edge locations
"

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 19 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:62]: I think that's clear.  Do you have a suggestion for rewording?

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 20 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:63]: replaced as suggested - thank you

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 21 above)
Comment (clubanderson) [reply docs/content/en/main/concepts/investigations/edge-multi-cluster.md:79]: added as suggested - thank you

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md (hunk 16 above)
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:55]: A. I suspect there is a typo here, I see the word "user" twice.
B. I do not think there is only one use case.  We have distilled some common technical problems from looking at many use cases.  Various hierarchies.  Various arrangements of roles and responsibilities.  Various security this-and-that.  Various sorts of testing.  "workload" and/or "policy".  Various degrees of intermittent connectivity.  Various sorts of testing.

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+   community contribution powered by integration and proper security features. All of these considerations must work across an evolving
+    hierarchy that combines small and large control plane API services.
+
+## Hierarchy and Scale are core considerations
+
+KCP has introduced the concept of a minimal API server which can be used to expand the number of places workloads can be deployed
+  by reducing the size of the control plane to a single binary. Furthermore, KCP's Transparent Multi-cluster (TMC) has made it possible to define
+  and deploy workloads centrally and then fan them out to specific edge locations. We would 
+   like to leverage different topologies (for instance: cascading multiple instances of KCP and Kubernetes cluster) to arrive at a 
+    distributed system capable of reaching any targeted edge cluster with/without edge devices attached. Topology flexibility
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:48]: The text does not make clear what is and is not meant by "cluster" (e.g., did not mention, let alone give proper emphasis to, the idea of single-node clusters) or "device".

File: docs/content/en/main/concepts/investigations/edge-multi-cluster.md [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+  situations when pods must run on a specific edge location on a specific node. Managing a set of labels, taints, and tolerations to cover all edge 
+  locations is too difficult.
+
+A key area of investigation for `kcp` is exploring large-scale deployment of workloads that can operate under constraints and
+ requirements unlike those accomodated by upstream Kubernetes or KCP's Transparent MultiCluster (TMC). Ideally, we want declaritively defined workloads 
+ to operate across a flexible topology combining KCP workspaces and Kubernetes cluster namespaces to achieve greater scale and control over placement. 
+ Deployment of workloads across many thousands of destinations that include namespaced and non-namespaced objects, receiving state and reporting
+  status (grouped and individually), require new interfaces based on (and extending) KCP primitives.
+
+## Goal: Placement strategies
```
Comment (MikeSpreitzer) [inline docs/content/en/main/concepts/investigations/edge-multi-cluster.md:27]: A. Why is this 2nd-level heading a "goal" and the next two _not_ a goal?
B. We should give peer handling of all the distinctive concerns of edge-MC.
